pkgs/desktops/gnome/misc/gnome-extensions-cli/default.nix

```diff
@@ -0,0 +1,42 @@
+{ lib
+, python3
+, fetchFromGitHub
+}:
+
+with python3.pkgs;
+
+buildPythonApplication rec {
+  pname = "gnome-extensions-cli";
+  version = "0.9.5";
+  format = "pyproject";
+
+  src = fetchPypi {
+    pname = "gnome_extensions_cli";
+    inherit version;
+    hash = "sha256-4eRVmG5lqK8ql9WpvXsf18znOt7kDSnpQnLfy73doy4=";
+  };
+
+  nativeBuildInputs = [
+    poetry-core
+  ];
```

Comment:
Do you think you could get the upstream tests to run using `pytestCheckHook` in `nativeBuildInputs`?

Reply:
Possibly but I don't know if that will work with CI since it looks like the tests require it to interact with a GNOME environment on the local machine via dbus. If one isn't up and running I'd imagine it would fail.

Reply:
https://github.com/essembeh/gnome-extensions-cli/blob/main/tests/test_dbus.py

Reply:
Actually, I see that it has a skip if not available. Maybe I can get that working.

@pytest.mark.skipif(not test_dbus_available(True), reason="DBus is not available")

Reply:
Hmm. That is failing.

Executing pytestCheckPhase
============================= test session starts ==============================
platform linux -- Python 3.10.10, pytest-7.2.1, pluggy-1.0.0
rootdir: /build/gnome_extensions_cli-0.9.5
collected 0 items                                                              

============================ no tests ran in 0.01s =============================
/nix/store/pw17yc3mwmsci4jygwalj8ppg0drz31v-stdenv-linux/setup: line 1593: pop_var_context: head of shell_variables not a function context
error: builder for '/nix/store/95g2b3lqh6w76ybp0lsswpy3i1lharn0-gnome-extensions-cli-0.9.5.drv' failed with exit code 5;
       last 10 log lines:
       > Check whether the following modules can be imported: gnome_extensions_cli
       > pytestCheckPhase
       > Executing pytestCheckPhase
       > ============================= test session starts ==============================
       > platform linux -- Python 3.10.10, pytest-7.2.1, pluggy-1.0.0
       > rootdir: /build/gnome_extensions_cli-0.9.5
       > collected 0 items                                                              
       >
       > ============================ no tests ran in 0.01s =============================
       > /nix/store/pw17yc3mwmsci4jygwalj8ppg0drz31v-stdenv-linux/setup: line 1593: pop_var_context: head of shell_variables not a function context
       For full logs, run 'nix log /nix/store/95g2b3lqh6w76ybp0lsswpy3i1lharn0-gnome-extensions-cli-0.9.5.drv'.

Reply:
I'm not sure how to fix this one myself, but I'm open to accepting help on it.

Reply:
Oh it's probably because you are fetching from PyPI instead of from GitHub. PyPI tarballs are often missing tests if they are separate from the package folders. If this doesn't get tests, it's not that big of a deal though (just always nice to have of course)

Reply:
This is a good call, I looked at the tarball contents and it indeed doesn't have the tests in it.